homeassistant/components/recorder/__init__.py

```diff
@@ -98,6 +99,9 @@
                     vol.Coerce(int), vol.Range(min=0)
                 ),
                 vol.Optional(CONF_DB_URL): cv.string,
+                vol.Optional(CONF_COMMIT_INTERVAL, default=1): vol.All(
```

Comment:
Since we write on shutdown, wouldn't it make sense to put this a lot higher? We could potentially also put a cap at max pending objects. So either every 30 seconds or 500 objects, whichever comes first ?

Reply:
My first thought was to be more aggressive with this and use the 50s I've been running at :)   I went with 1s since I was trying to manage change risk and this had the most return for the least amount of change in testing.   

How do you feel about keeping it at 1 for now, and adjusting the default to something higher + adding the max commit backlog after the beta cut so if a higher value causes any unforeseen issues it will have more time to shake out?

Reply:
Sounds good.

Reply:
To close the loop on this, I've been testing various commit intervals over the last few months, and 1s seems to be the sweet spot as anything else is diminishing returns and a delay with logbook/history graphs.